.github/workflows/build_and_deploy_ext2_img.yml

```diff
@@ -0,0 +1,119 @@
+        description: 'Path to the Dockerfile.'
+        required: true
+        default: 'dockerfiles/Dockerfile'
+  
+      image_size:
+        description: 'Size of the base ext2 image. 1G is the max and recommended size. Reference the dd manpage for multiplicative suffixes'
+        required: true
+        default: '1G'
+
+      image_name:
+        description: 'Name of final image. Example final url: https://www.github.io/<owner>/<repositoryName>/<image_name>'
+        required: true
+        default: 'base_image.ext2'
+
+      deploy_to_pages:
+        description: 'Set to true to deploy to pages.'
+        required: true
+        default: 'true'
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest # Image to run the worker on.
+    
+    # Our variables
+    env:
+      ext2_img: "ext2-webvm-base-image"
+      dockerfile_context_path: '${{ github.event.inputs.dockerfile_context_path }}'
+      dockerfile_path: '${{ github.event.inputs.dockerfile_path }}'
+
+      tag: "ext2_img"
+      IMAGE_SIZE: '${{ github.event.inputs.image_size }}'
+      IMAGE_NAME: '${{ github.event.inputs.image_name }}'
+      DEST: /assets/images/tmp/ # Path to directory where we host the final image from.
+  
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so our job can access it
+      - uses: actions/checkout@v3
+      # We make the directory to host our image from.
+      - run: sudo mkdir -p /assets/images/tmp/
+      # We build the i386 Dockerfile image.
+      - run: docker build $dockerfile_context_path --file $dockerfile_path --tag $tag --platform=i386 
+      # We run our image with tail -f /dev/null to keep it running.
+      - run: docker run --rm -d $tag tail -f /dev/null
+      # We create our ext2 filesystem.
+      # We allocate space for the ext2 image
+      - run: sudo dd if=/dev/zero of=${DEST}${IMAGE_NAME} bs=1 count=0 seek=$IMAGE_SIZE
+      # Format to ext2 linux kernel revision 0
+      - run: sudo mkfs.ext2 -r 0 -F ${DEST}${IMAGE_NAME}
+      # We mount the ext2 image in order to modify it.
+      - run: | 
+          sudo mkdir -p /mnt/ext2_image
+          sudo mount -o loop -t ext2 ${DEST}/${IMAGE_NAME} /mnt/ext2_image
+      # We export our container its filesystem and unpack it into the mounted ext2 image.
+      - run: docker export $(sudo docker ps -aq) | sudo tar -x -C /mnt/ext2_image
+      # We kill the container.
+      - run: |
+          docker kill $(docker ps -a -q)
+          sudo umount /mnt/ext2_image
+      
+      # Result is a nice ext2 image for webvm.
+      - name: Upload GitHub Pages artifact
+        uses: actions/upload-pages-artifact@v1.0.7
+        with:
+          # Artifact name
+          name: github-pages # optional, default is github-pages
+          # Path of the directory containing the static assets.
+          path: /assets/images/tmp/ # default is _site/
+          # Duration after which artifact will expire in days.
+          retention-days: 1 # optional, default is 1
```

Comment:
Can we allow retention till the next build?

Reply:
Unfortunately not possible, retention-days can only be between 1 to 90 days for public repositories. Should I set it to 90 days?

https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization

Reply:
Can we publish the result as a "release"?